src/app/core/services/postulants.service.ts

```diff
@@ -15,6 +15,13 @@ export class PostulantsService extends DataService<Postulant> {
     super(db, FirestoreCollection.attendees);
   }
 
+  getIsPostulantExist(email: string): Promise<boolean> {
```

Comment:
All services extend from `DataService`, which already has a function to recover data by id, I suggest that you modify that one a little so it can jump the `deleted` validation in some scenarios like this one and use it. In `DataService`:
```
  getById(id: string, validateDeletedFlag = true): Observable<T> {
    return this.db
      .doc<T>(`${this.collection}/${id}`)
      .valueChanges()
      .pipe(
        map((data) => {
          if ((validateDeletedFlag && !data.deleted) || !validateDeletedFlag) {
            return data;
          }

          return null;
        }),
      );
  }
```

And in the `auth.service` you can use it like this:
```
  login(email: string, password: string): void {
    this.loading = true;

    this.afAuth
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        this.message = '';

        this.postulantsService
          .getById(user.uid)
          .pipe(
            switchMap((postulant) => {
              if (!postulant) {
                return this.userService.assertAuthUser(user);
              }

              throw new Error('This user does not have permission to access this site.');
            }),
            first(),
          )
          .subscribe((_) => this.router.navigate(['']));

        timer(500).subscribe(() => (this.loading = false));
      })
      .catch((err) => {
        this.message = err;
        this.loading = false;
      });
  }
```